### PR TITLE
machines: Don't create VM with name which is already used

### PR DESCRIPTION
**Pull request by skobyda**
When user creates a new VM and spends some time filling in all the details, after clicking Create, modal is closed with error message from libvirt "this vm name is already used" resulting in them loosing all the data they have filled into the modal.
We can check this in the UI.

Screenshot from 2019-11-07 15-26-57

### CODE REVIEW

File: pkg/machines/components/create-vm-dialog/createVmDialog.jsx [hunk 1]
```diff
@@ -116,11 +116,13 @@ function getSpaceAvailable(storagePools, connectionName) {
...
 }
 
-function validateParams(vmParams) {
+function validateParams(vmParams, vms) {
     const validationFailed = {};
 
     if (isEmpty(vmParams.vmName.trim()))
         validationFailed.vmName = _("Name must not be empty");
+    else if (vms.some(vm => vm.name === vmParams.vmName))
+        validationFailed.vmName = _("VM ") + vmParams.vmName + _(" already exists");
```
Comment (KKoukiou) [inline pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: You can use cockpit.format for that,
cockpit.format(_("VM $0 already exists"), vmParams.vmName);

I think it looks nicer tbh. And tranlations won't be weird string finishing/starting with spaces.

File: pkg/machines/components/create-vm-dialog/createVmDialog.jsx [hunk 2]
```diff
@@ -732,7 +734,7 @@ class CreateVmModal extends React.Component {
 
     render() {
         const { nodeMaxMemory, nodeDevices, networks, osInfoList, loggedUser, providerName, storagePools, vms } = this.props;
-        const validationFailed = this.state.validate && validateParams({ ...this.state, osInfoList });
+        const validationFailed = this.state.validate && validateParams({ ...this.state, osInfoList }, vms.filter(vm => vm.connectionName == this.state.connectionName));
```
Comment (KKoukiou) [inline pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: Why you don't keep the new paramers inside the dict to allow it be named parameter?
Then you can access it with vmParams.vms from the function.
It's just weird to have some args names and some non.

File: test/verify/machineslib.py [hunk 3]
```diff
@@ -1368,10 +1368,20 @@ def installWithErrorTest(dialog):
 
         checkFilteredOsTest(TestMachines.VmDialog(self, storage_size=1, os_name=config.MAGEIA_3_FILTERED_OS))
 
+
```
Comment (KKoukiou) [inline test/verify/machineslib.py]: Unrelated newlines change.

File: test/verify/machineslib.py [hunk 4]
```diff
@@ -1368,10 +1368,20 @@ def installWithErrorTest(dialog):
...
         checkDialogFormValidationTest(TestMachines.VmDialog(self, "", storage_size=1), {"Name": "Name must not be empty"})
 
+        # name already exists
+        createTest(TestMachines.VmDialog(self, name='existing-name', sourceType='url',
+                                         location=config.VALID_URL, storage_size=1,
+                                         delete=False))
+
+        checkDialogFormValidationTest(TestMachines.VmDialog(self, "existing-name", storage_size=1), {"Name": "already exists"})
+
+        self.machine.execute("virsh destroy existing-name")
```
Comment (KKoukiou) [inline test/verify/machineslib.py]: This will not undefine the VM though, it will still appear on the list and it will be annoying when debugging future test failures.

File: pkg/machines/components/create-vm-dialog/createVmDialog.jsx (hunk 1 above)
Comment (martinpitt) [reply pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: Moreover, we must not enforce word ordering in translatable strings, other languages may put the parameter at a different position.

File: pkg/machines/components/create-vm-dialog/createVmDialog.jsx (hunk 2 above)
Comment (skobyda) [reply pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: Oh I tought that `vmParams` refers to just dialog state. Now I noticed it also contais osInfoList, so yeah, it would be fine to put it there.

File: test/verify/machineslib.py [hunk 5]
```diff
@@ -1372,6 +1373,16 @@ def installWithErrorTest(dialog):
...
 
+        # name already exists
+        createTest(TestMachines.VmDialog(self, name='existing-name', sourceType='url',
+                                         location=config.VALID_URL, storage_size=1,
+                                         delete=False))
+
+        checkDialogFormValidationTest(TestMachines.VmDialog(self, "existing-name", storage_size=1,
+                                                            env_is_empty=False), {"Name": "already exists"})
+
+        self.machine.execute("virsh destroy existing-name && virsh undefined existing-name")
```
Comment (KKoukiou) [inline test/verify/machineslib.py]: Typo here - virsh undefine is the correct command.
